main.go

```diff
@@ -426,7 +459,12 @@ func main() {
 				fail("Failed to write export options to file, error: %s", err)
 			}
 		} else {
-			exportOptionsContent, err := generateExportOptionsPlist(configs.ExportMethod, configs.TeamID, configs.UploadBitcode, configs.CompileBitcode, xcodebuildVersion.MajorVersion, archive)
+			productToDistribute, err := ParseExportProduct(configs.ProductToDistribute)
```

Comment:
It works better in my experience if you parse the input at the very first, when step configs are read.